src/mono/mono/mini/interp/interp.c

```diff
@@ -1466,9 +1466,13 @@ build_args_from_sig (MonoMethodSignature *sig, InterpFrame *frame)
 
 #ifdef HOST_WASM
 			{
+				MonoType *etype;
+
 				/* Scalar vtypes are passed by value */
-				if (mini_wasm_is_scalar_vtype (sig->params [i]))
+				// FIXME: r4/r8
+				if (mini_wasm_is_scalar_vtype (sig->params [i], &etype) && etype->type != MONO_TYPE_R4 && etype->type != MONO_TYPE_R8) {
 					margs->iargs [int_i] = *(gpointer*)margs->iargs [int_i];
```

Comment:
Will this work correctly for i8/u8? I'm fine with not fixing those yet though

Reply:
We disable those:
		} else if (size == 8 && t->type != MONO_TYPE_R8) {
			return FALSE;